### PR TITLE
Add Optional Auth to BFD protocol

### PR DESCRIPTION
**Pull request by KaptainH**
The BFD module of Scapy (contrib/bfd.py) does not support Optional Authentification as described in https://www.rfc-editor.org/rfc/rfc5880#section-6.7

1. New class OptionalAuth is added to support all type of Authentification.
2. The "len" field of the BFD class is switched from ByteField to FieldLenField in order to support variable length in case of auth is used.
3. A ConditionnalField is added to the BFD class to add an OptionalAuth PacketField if the flag A is set.
4. Original uts file is improved

### CODE REVIEW

File: scapy/contrib/bfd.py [hunk 1]
```diff
@@ -58,10 +143,12 @@ def mysummary(self):
...
-for _bfd_port in [3784,          # single-hop BFD
-                  4784,          # multi-hop BFD
-                  6784,          # BFD for LAG a.k.a micro-BFD
-                  7784]:         # seamless BFD
+for _bfd_port in [
+    3784,  # single-hop BFD
+    4784,  # multi-hop BFD
+    6784,  # BFD for LAG a.k.a micro-BFD
+    7784,
+]:  # seamless BFD
```
Comment (gpotter2) [inline scapy/contrib/bfd.py:151]: nit: the comment is misplaced

File: scapy/contrib/bfd.py [hunk 2]
```diff
@@ -35,20 +52,88 @@
...
+class OptionalAuth(Packet):
+    name = "Optional Auth"
+    fields_desc = [
+        ByteEnumField("auth_type", 1, _authentification_type),
+        FieldLenField(
+            "auth_len",
+            None,
+            fmt="B",
+            length_of="auth_key",
+            adjust=lambda pkt, x: int(x + 3) if pkt.auth_type <= 1 else int(x + 8),
```
Comment (gpotter2) [inline scapy/contrib/bfd.py:75]: nit: what do you need the `int(...)` for?

**Review by gpotter2 — COMMENTED** [review]
Hi & thanks for the PR !
Looking good, just a few comments:

File: scapy/contrib/bfd.py (hunk 2 above)
Comment (KaptainH) [reply scapy/contrib/bfd.py:75]: Don't need thoses int(...) at all '^^

File: scapy/contrib/bfd.py (hunk 1 above)
Comment (KaptainH) [reply scapy/contrib/bfd.py:151]: Oh did not even noticed code formating tool touch this one...

**Comment by gpotter2** [comment]
Hi ! Could you fix the small PEP8 mistake? (see the first failed test) thanks !

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**